frontend/src/components/core/Sidebar/SidebarNav.tsx

```diff
@@ -57,7 +57,18 @@ const SidebarNav = ({
     return null
   }
 
-  const [expanded, setExpanded] = useState(false)
+  // Check localStorage to see if the user has a preference set
+  const isLocalStorageSet = localStorage.getItem("navExpanded") !== null
```

Comment:
This [recently merged PR](https://github.com/streamlit/streamlit/pull/6100) has implications for usage of `localStorage` -> essentially the need to check its availability as a precursor.

Reply:
Oh, I see! Thanks for letting me know, updated on [8fef855](https://github.com/streamlit/streamlit/pull/6050/commits/8fef8554a2d7db7e0d1eef487f8aea15e2b50efc)